openeo_processes_dask/process_implementations/cubes/apply.py

```diff
@@ -31,14 +32,43 @@ def apply_dimension(
     dimension: str,
     target_dimension: Optional[str] = None,
     context: Optional[dict] = None,
-    **kwargs,
 ) -> RasterCube:
+    if context is None:
+        context = {}
+
     if dimension not in data.dims:
         raise DimensionNotAvailable(
             f"Provided dimension ({dimension}) not found in data.dims: {data.dims}"
```

Comment:
In the documentation the DimensionNotAvailable Message is specified as: "A dimension with the specified name does not exist." Anyway, it makes sense to explicitly list the available dimensions here, so maybe we can just stick to this error message. :)

Reply:
Yeah, we currently deviate from the specified messages in a few places, where it makes sense to report more details, think that's fine tbh!